app/Phragile/TransactionRawDataProcessor.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Phragile;
+
+class TransactionRawDataProcessor {
+
+	public function process(array $rawData)
+	{
+		$transactions = [];
+		foreach ($rawData as $task => $taskTransactionData)
+		{
+			$transactions[$task] = $this->processTaskTransactionData($taskTransactionData);
+		}
+		return $transactions;
+	}
+
+	private function processTaskTransactionData(array $rawTaskTransactionData)
+	{
+		$transactions = [];
+		foreach ($rawTaskTransactionData as $singleTransactionData)
+		{
+			if ($this->isColumnChangeTransaction($singleTransactionData))
+			{
+				$transactions[] = new ColumnChangeTransaction([
+					'timestamp' => $this->getTransactionTimestamp($singleTransactionData),
+					'workboardPHID' => $this->getWorkboardPHID($singleTransactionData),
+					'oldColumnPHID' => $this->getOldColumnPHID($singleTransactionData),
+					'newColumnPHID' => $this->getNewColumnPHID($singleTransactionData),
+				]);
+			} elseif ($this->isStatusChangeTransaction($singleTransactionData))
+			{
+				$transactions[] = new StatusChangeTransaction([
+					'timestamp' => $this->getTransactionTimestamp($singleTransactionData),
+					'oldStatus' => $this->getOldStatus($singleTransactionData),
+					'newStatus' => $this->getNewStatus($singleTransactionData),
+				]);
+			} elseif ($this->isMergedIntoTransaction($singleTransactionData))
+			{
+				$transactions[] = new MergedIntoTransaction(
+					$this->getTransactionTimestamp($singleTransactionData)
+				);
+			}
+		}
+		return $transactions;
+	}
+
+	private function isColumnChangeTransaction(array $rawData)
+	{
+		return array_key_exists('transactionType', $rawData)
+			&& $rawData['transactionType'] === 'core:columns';
```

Comment:
oh, of course the phrabricator transaction type string is also used explicitly in `TransactionFilter`, that's what you meant, sorry! Forget what I was mumbling about in the above comment.
Then moving the string to a constant makes sense. But I'd definitely keep it away from `*Transaction` classes